Helper/ConfigHelper.php

```diff
@@ -476,6 +476,10 @@ public function getSortingIndices($originalIndexName, $storeId = null)
                 foreach ($groupCollection as $group) {
                     $customerGroupId = (int) $group->getData('customer_group_id');
 
+                    if (!is_null($currentCustomerGroupId) && $customerGroupId != $currentCustomerGroupId) {
```

Comment:
First of all - the method is super messy and worth complete refactoring. But for the simplicity sake - you are right, @bsuravech. It can be simplified. But `getItemById()` loads the full collection anyway. I'd prefer add a filter condition between creation of the collection and the `foreach`.

So move the condition `if (!is_null($currentCustomerGroupId)) { ... }` to line 475 and add filter `$groupCollection->addFilter('id', $currentCustomerGroupId);` there. In the end:

```
if (!is_null($currentCustomerGroupId)) {
    $groupCollection->addFilter('id', $currentCustomerGroupId);
}
```

Would that work @damcou? We would save the load of the entire collection and at the same time we wouldn't iterate over the full collection if it's not needed. And we wouldn't have to rewrite the full `foreach`.

WDYT?

Reply:
Done! This is indeed better this way ;) .
And I agree that this method is a good candidate for a future refactoring .